CRM/Activity/BAO/Activity.php

```diff
@@ -2140,6 +2140,10 @@ public static function getActivitySubject($entityObj) {
         $membershipType = CRM_Member_PseudoConstant::membershipType($entityObj->membership_type_id);
         $subject = $membershipType ? $membershipType : ts('Membership');
```

Comment:
could it be instead 
  $subject = $membershipType ? (string) $membershipType : ts('Membership');